src/components/Fields/TextField.test.js

```diff
@@ -363,4 +363,19 @@ describe('TextField', () => {
         const textField = shallow(<TextField name={exampleName} />);
         expect(textField.find('input').prop('name')).to.equal(exampleName);
     });
+
+    it('adds --has-value when value is not null', () => {
+        const textField = shallow(<TextField value="test" />);
+        expect(textField.hasClass('uir-text-field--has-value')).to.equal(true);
+    });
+
+    it('adds --has-value when value is falsy but not null', () => {
+        const textField = shallow(<TextField value={0} />);
+        expect(textField.hasClass('uir-text-field--has-value')).to.equal(true);
+    });
+
+    it('does not add --has-value when value isnull', () => {
```

Comment:
Technically, this should say "when value is undefined". If you try setting the `value` to `null` it will actually add the `--has-value` class (but React will also log an error in the console about using `null`).

Reply:
Ahh yes, good point